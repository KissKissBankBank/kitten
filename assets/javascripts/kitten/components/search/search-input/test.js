import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { SearchInput } from './index'

describe('<SearchInput />', () => {
  let component

  describe('Basic search input', () => {
    beforeEach(() => {
      component = renderer.create(<SearchInput />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('Search input with attributes', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <SearchInput
            aria-label="Rechercher"
            placeholder="Rechercher"
            className="k-testClass"
          />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
