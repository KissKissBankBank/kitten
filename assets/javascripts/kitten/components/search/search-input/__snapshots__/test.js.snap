// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> Basic search input matches with snapshot 1`] = `
.c0 {
  width: 100%;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: Maax,Helvetica,Arial,sans-serif;
  -webkit-letter-spacing: .01rem;
  -moz-letter-spacing: .01rem;
  -ms-letter-spacing: .01rem;
  letter-spacing: .01rem;
  font-weight: 600;
  caret-color: #19b4fa;
  font-size: 1.75rem;
  line-height: 2.75rem;
}

.c0::-webkit-input-placeholder {
  color: #eee;
}

.c0::-moz-placeholder {
  color: #eee;
}

.c0:-ms-input-placeholder {
  color: #eee;
}

.c0::placeholder {
  color: #eee;
}

.c0::-webkit-search-decoration,
.c0::-webkit-search-cancel-button,
.c0::-webkit-search-results-button,
.c0::-webkit-search-results-decoration {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

@media (min-width:640px) {
  .c0 {
    font-size: 3rem;
    line-height: 4.25rem;
  }
}

@media (min-width:1080px) {
  .c0 {
    font-size: 4rem;
    line-height: 5.5rem;
  }
}

<input
  className="c0"
  type="search"
/>
`;

exports[`<SearchInput /> Search input with attributes matches with snapshot 1`] = `
.c0 {
  width: 100%;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: Maax,Helvetica,Arial,sans-serif;
  -webkit-letter-spacing: .01rem;
  -moz-letter-spacing: .01rem;
  -ms-letter-spacing: .01rem;
  letter-spacing: .01rem;
  font-weight: 600;
  caret-color: #19b4fa;
  font-size: 1.75rem;
  line-height: 2.75rem;
}

.c0::-webkit-input-placeholder {
  color: #eee;
}

.c0::-moz-placeholder {
  color: #eee;
}

.c0:-ms-input-placeholder {
  color: #eee;
}

.c0::placeholder {
  color: #eee;
}

.c0::-webkit-search-decoration,
.c0::-webkit-search-cancel-button,
.c0::-webkit-search-results-button,
.c0::-webkit-search-results-decoration {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

@media (min-width:640px) {
  .c0 {
    font-size: 3rem;
    line-height: 4.25rem;
  }
}

@media (min-width:1080px) {
  .c0 {
    font-size: 4rem;
    line-height: 5.5rem;
  }
}

<input
  aria-label="Rechercher"
  className="c0 k-testClass"
  placeholder="Rechercher"
  type="search"
/>
`;
